File: dynabuffers-java/src/main/kotlin/dynabuffers/ast/datatype/LongType.kt
package dynabuffers.ast.datatype

import dynabuffers.api.IRegistry
import dynabuffers.api.ISerializable
import dynabuffers.api.IType
import java.nio.ByteBuffer

class LongType : IType, ISerializable {

    override fun size(value: Any?, registry: IRegistry) = 8

    override fun serialize(value: Any?, buffer: ByteBuffer, registry: IRegistry) {
        buffer.putLong(long(value))
    }

    override fun deserialize(buffer: ByteBuffer, registry: IRegistry) = buffer.long

}

File: dynabuffers-python/dynabuffers/ast/datatype/LongType.py
from dynabuffers.api.ISerializable import ISerializable


class LongType(ISerializable):

    def size(self, value, registry):
        return 8

    def serialize(self, value, buffer, registry):
        buffer.putLong(value if isinstance(value, int) else int(value))

    def deserialize(self, buffer, registry):
        return buffer.getLong()
